src/core/qgsdataitem.h

```diff
@@ -143,6 +143,17 @@ class CORE_EXPORT QgsDataItem : public QObject
      */
     virtual QList<QAction *> actions() { return QList<QAction *>(); }
 
+    /** Returns the list of menus available for this item. This is usually used for the popup menu on right-clicking
+     * the item. Subclasses should override this to provide actions. Ownership of the returned menus is unchanged.
```

Comment:
You'll just need to update these docs now:

"Subclasses should ensure that ownership of created menus is correctly handled by parenting them to the specified \a parent widget"